homeassistant/components/octoprint/camera.py

```diff
@@ -70,3 +108,152 @@ def __init__(
             unique_id=device_id,
             verify_ssl=verify_ssl,
         )
+
+
+class _OctoprintCamera(CoordinatorEntity[OctoprintDataUpdateCoordinator], Camera):
+    """Representation of an OctoPrint Camera Stream."""
+
+    _attr_name = "OctoPrint Camera"
+    _attr_supported_features = CameraEntityFeature.STREAM
+    _attr_is_streaming = True
+
+    def __init__(
+        self,
+        camera_settings: WebcamSettings,
+        coordinator: OctoprintDataUpdateCoordinator,
+        device_id: str,
+        verify_ssl: bool,
+    ) -> None:
+        CoordinatorEntity.__init__(
+            self,
+            coordinator=coordinator,
+        )
+        Camera.__init__(self)
+
+        self._attr_device_info = coordinator.device_info
+        self._attr_unique_id = device_id
+
+        self._camera_settings = camera_settings
+        self._verify_ssl = verify_ssl
+
+        self._last_image: bytes | None = None
+        self._last_update = datetime.min
+        self._update_lock = asyncio.Lock()
+
+    @cached_property
+    def use_stream_for_stills(self) -> bool:
+        """Use stream for stills if no snapshot URL is available."""
+        return not self._camera_settings.internal_snapshot_url
+
+    async def async_camera_image(
+        self, width: int | None = None, height: int | None = None
+    ) -> bytes | None:
+        "Fetch the camera image from the Octoprint URL."
+
+        if not (url := self._camera_settings.internal_snapshot_url):
+            return None
+
+        async with self._update_lock:
+            if (
+                self._last_image is not None
+                and self._last_update + timedelta(0, self._attr_frame_interval)
+                > datetime.now()
+            ):
+                return self._last_image
+
+            try:
+                update_time = datetime.now()
+                async_client = get_async_client(
+                    self.hass,
+                    verify_ssl=self._verify_ssl
+                    and self._camera_settings.snapshot_ssl_validation_enabled,
+                )
+                response = await async_client.get(
+                    url,
+                    follow_redirects=True,
+                    timeout=GET_IMAGE_TIMEOUT,
+                )
+                response.raise_for_status()
+                self._last_image = response.content
```

Comment:
Consider adding validation of `response.content`

Reply:
To maybe pull this one further, this is a device specific call, and that should belong in the library